knowledge-base/scheduler-kb-context-menu-on-appointments-and-empty-slots.md

```diff
@@ -0,0 +1,367 @@
+                    <div style="display:flex; width: 100%; height:100%"
+                         @oncontextmenu:preventDefault="true"
+                         @oncontextmenu="@( (MouseEventArgs e) => ShowContextMenuFromEmptySlot(e, EmptySlot) )">
+                    </div>
+                }
+            </SlotTemplate>
+        </SchedulerWeekView>
+        <SchedulerMultiDayView StartTime="@DayStart" NumberOfDays="10">
+            <SlotTemplate>
+                @{
+                    SchedulerSlotTemplateContext EmptySlot = (SchedulerSlotTemplateContext)context;
+                    <div style="display:flex; width: 100%; height:100%"
+                         @oncontextmenu:preventDefault="true"
+                         @oncontextmenu="@( (MouseEventArgs e) => ShowContextMenuFromEmptySlot(e, EmptySlot) )">
+                    </div>
+                }
+            </SlotTemplate>
+        </SchedulerMultiDayView>
+    </SchedulerViews>
+    <ItemTemplate>
+        @{
+            SchedulerAppointment Appointment = (SchedulerAppointment)context;
+            <div style="height:100%" class="@( Appointment.IsImportant ? "important-appt" : "" )"
+                 @oncontextmenu:preventDefault="true"
+                 @oncontextmenu="@( (MouseEventArgs e) => ShowItemContextMenu(e, Appointment) )">
+                <div style="height:100%" class="k-event-template">@Appointment.Title</div>
+            </div>
+        }
+    </ItemTemplate>
+    <AllDayItemTemplate>
+        @{
+            SchedulerAppointment Appointment = (SchedulerAppointment)context;
+            <div style="height:100%" class="@( Appointment.IsImportant ? "important-appt" : "" )"
+                 @oncontextmenu:preventDefault="true"
+                 @oncontextmenu="@( (MouseEventArgs e) => ShowItemContextMenu(e, Appointment) )">
+                <div style="height:100%" class="k-event-template">@Appointment.Title</div>
+            </div>
+        }
+    </AllDayItemTemplate>
+</TelerikScheduler>
+
+@* ContextMenu for ItemTemplate and AllDayItemTemplate *@
+<TelerikContextMenu @ref="@TheItemContextMenu"
+                    Data="@MenuItems"
+                    TextField="Text"
+                    IconField="Icon"
+                    DisabledField="Disabled"
+                    OnClick="@( async (ContextMenuItem itm) => await MenuClickItemHandler(itm) )">
+</TelerikContextMenu>
+
+@* ContextMenu for SlotTemplate *@
+<TelerikContextMenu @ref="@TheSlotContextMenu"
+                    Data="@MenuSlots"
+                    TextField="Text"
+                    IconField="Icon"
+                    DisabledField="Disabled"
+                    OnClick="@( async (ContextMenuItem itm) => await MenuClickSlotHandler(itm) )">
+</TelerikContextMenu>
+
+<style>
+    .important-appt {
+        color: purple;
+        font-weight: bold;
+        font-size: 1.5em;
+    }
+</style>
+
+@code {
+    private TelerikScheduler<SchedulerAppointment> SchedulerRef { get; set; }
```

Comment:
Try to produce code that doesn't generate warnings. This change will required some more where `SchedulerRef` is used.
```suggestion
    private TelerikScheduler<SchedulerAppointment>? SchedulerRef { get; set; }
```